cmd/cri-containerd/options/options.go

```diff
@@ -105,63 +104,52 @@ func NewCRIContainerdOptions() *CRIContainerdOptions {
 	fs.StringVar(&c.NetworkPluginConfDir, "network-conf-dir",
-		"/etc/cni/net.d", "The directory for putting network plugin configuration files.")
+		defaults.NetworkPluginConfDir, "The directory for putting network plugin configuration files.")
 	fs.StringVar(&c.StreamServerAddress, "stream-addr",
-		"", "The ip address streaming server is listening on. The default host interface is used if not specified.")
+		defaults.StreamServerAddress, "The ip address streaming server is listening on. The default host interface is used if not specified.")
 	fs.StringVar(&c.StreamServerPort, "stream-port",
-		"10010", "The port streaming server is listening on.")
+		defaults.StreamServerPort, "The port streaming server is listening on.")
 	fs.StringVar(&c.CgroupPath, "cgroup-path",
-		"", "The cgroup that cri-containerd is part of. Cri-containerd is not placed in a cgroup if none is specified.")
+		defaults.CgroupPath, "The cgroup that cri-containerd is part of. Cri-containerd is not placed in a cgroup if none is specified.")
 	fs.BoolVar(&c.EnableSelinux, "enable-selinux",
-		false, "Enable selinux support. By default not enabled.")
+		defaults.EnableSelinux, "Enable selinux support. By default not enabled.")
 	fs.StringVar(&c.SandboxImage, "sandbox-image",
-		"gcr.io/google_containers/pause:3.0", "The image used by sandbox container.")
+		defaults.SandboxImage, "The image used by sandbox container.")
 	fs.IntVar(&c.StatsCollectPeriod, "stats-collect-period",
-		10, "The period (in seconds) of snapshots stats collection.")
+		defaults.StatsCollectPeriod, "The period (in seconds) of snapshots stats collection.")
 	fs.BoolVar(&c.SystemdCgroup, "systemd-cgroup",
-		false, "Enables systemd cgroup support. By default not enabled.")
-	fs.BoolVar(&c.PrintDefaultConfig, "default-config",
-		false, "Print default toml config of cri-containerd and quit.")
+		defaults.SystemdCgroup, "Enables systemd cgroup support. By default not enabled.")
 	fs.IntVar(&c.OOMScore, "oom-score",
-		-999, "Adjust the cri-containerd's oom score.")
+		defaults.OOMScore, "Adjust the cri-containerd's oom score.")
 }
 
-// InitFlags must be called after adding all cli options flags are defined and
-// before flags are accessed by the program. Ths fuction adds flag.CommandLine
-// (the default set of command-line flags, parsed from os.Args) and then calls
-// pflag.Parse().
+// InitFlags load configurations from config file, and then overwrite with flags.
+// This function must be called inside `Run`, at that time flags should have been
+// parsed once.
 // precedence:  commandline > configfile > default
 func (c *CRIContainerdOptions) InitFlags(fs *pflag.FlagSet) error {
-	fs.AddGoFlagSet(flag.CommandLine)
-
-	commandline := os.Args[1:]
-	if err := fs.Parse(commandline); err != nil {
-		return err
-	}
-
 	// Load default config file if none provided
 	if _, err := toml.DecodeFile(c.ConfigFilePath, &c.Config); err != nil {
```

Comment:
Can we modify ConfigFilePath to specification?

Reply:
What do you mean?

Reply:
You has answered it at below. Tks